src/lib/microservices/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,310 @@
+
+const debug: debug.IDebugger = d(__filename);
+const { QueueConnection, DatabaseConnection: dbConnectionString } = process.env; // eslint-disable-line no-process-env
+let queue: Queue = null;
+const moduleName: string = 'Scanner API';
+const categories = require('./categories.json');
+
+const connectToQueue = () => {
+    if (queue) {
+        return;
+    }
+
+    /* istanbul ignore else */
+    if (QueueConnection) {
+        queue = new Queue('webhint-jobs-dev', QueueConnection);
+    } else {
+        logger.log('Queue connection string not found', moduleName);
+    }
+};
+
+/**
+ * Split the job in as many messages as configurations it has.
+ * @param {IJob} job - Job to send to the queue.
+ */
+const sendMessagesToQueue = async (job: IJob) => {
+    connectToQueue();
+
+    let counter = 0;
+
+    logger.log(`Splitting the Job in ${job.config.length} tasks`, moduleName);
+
+    for (const config of job.config) {
+        const jobCopy = _.cloneDeep(job);
+
+        jobCopy.partInfo = {
+            part: ++counter,
+            totalParts: job.config.length
+        };
+        jobCopy.config = [config];
+        await queue.sendMessage(jobCopy);
+        logger.log(`Part ${jobCopy.partInfo.part} of ${jobCopy.partInfo.totalParts} sent to the Service Bus`, moduleName);
+    }
+};
+
+/**
+ * Get the right configuration for the job.
+ * @param {RequestData} data - The data the user sent in the request.
+ */
+const getConfig = (data: JobData, serviceConfig: IServiceConfig): Array<UserConfig> => {
+    const source: ConfigSource = data.source;
+    let config: Array<UserConfig>;
+
+    debug(`Configuration source: ${source}`);
+    switch (source) {
+        /* istanbul ignore next */
+        case ConfigSource.file:
+            config = Array.isArray(data.config) ? data.config : [data.config];
+            break;
+        /* TODO: TBD. */
+        // case ConfigSource.manual:
+        default:
+            config = serviceConfig.webhintConfigs;
+            break;
+    }
+
+    return config;
+};
+
+/**
+ * Get a Job that it is still valid.
+ * @param {Array<IJob>} jobs - All the jobs for that url in the database.
+ * @param config - Job configuration.
+ */
+const getActiveJob = (jobs: Array<IJob>, config: Array<UserConfig>, cacheTime: number) => {
+    return jobs.find((job) => {
+        // job.config in cosmosdb is undefined if the config saved was an empty object.
+        return _.isEqual(job.config || [{}], config) && job.status !== JobStatus.error && (job.status !== JobStatus.finished || moment(job.finished).isAfter(moment().subtract(cacheTime, 'seconds')));
+    });
+};
+
+/**
+ * Create a new Job in the database.
+ * @param {string} url - The url that the job will be use.
+ * @param {UserConfig} config - The configuration for the job.
+ */
+const createNewJob = async (url: string, configs: Array<UserConfig>, jobRunTime: number): Promise<IJob> => {
+    let hints: Array<Hint> = [];
+
+    configs.forEach((config) => {
```

Comment:
We need to take into account also the hints coming from an `extends`: https://github.com/webhintio/online-service/issues/589

Reply:
Can you provide more details, I am not sure what the extends stuff is, this is direct migration of createNewJob from online-scanner

Reply:
`UserConfig` has a property `extends` that property will load a `configuration-` and will use the rules defined in that configuration. We should have those hints into account. Anyways, can take care of this.